pylint_odoo/checkers/no_modules.py

```diff
@@ -479,11 +485,15 @@ def visit_print(self, node):
                           'translation-required',
                           'translation-contains-variable',
                           'print-used',
+                          'str-format-used',
                           )
     def visit_call(self, node):
         infer_node = utils.safe_infer(node.func)
-        if utils.is_builtin_object(infer_node) and infer_node.name == 'print':
-            self.add_message('print-used', node=node)
+        if utils.is_builtin_object(infer_node):
+            if self.get_func_name(node.func) == 'format':
+                self.add_message('str-format-used', node=node)
```

Comment:
Sorry because I don't understand fine this code, but are you checking the source of the string?

I mean, using `"This is a {}".format(thing)` is :+1:, but  `_("This is a {}").format(thing)` is :-1:. I would only forbid `.format()` when the pattern string is untrusted.

Reply:
This code is checking if `*.format` is used.

I didn't get you.

> I mean, using `"This is a {}".format(thing)` is :+1:, but  `_("This is a {}").format(thing)` is :-1:. I would only forbid `.format()` when the pattern string is untrusted.

I didn't get you.

Why `"This is a {}".format(thing)` is good?

Reply:
You can test locally this branch using the following steps:

```bash
git clone https://github.com/vauxoo-dev/pylint-odoo.git -b master-oca-avoid-format-moy /tmp/pylint-odoo
cd /tmp/pylint-odoo
python3 -m pip install -r requirements.txt
python3 -m pylint -d all -e str-format-used --load-plugins=pylint_odoo pylint_odoo/test_repo/
```

So, you can change the files in test_repo/* to test different cases

Reply:
> Why `"This is a {}".format(thing)` is good?

It is good because the template (`"This is a {}"`) is trusted.

If the template came from an untrusted source, such as a translation file or an argument in a public method, it would be dangerous.

Reply:
Hi @Yajo 

I have made some changes considering you mentioned.

Could you review again, please?

Regards.